probe-rs/src/architecture/arm/core/armv6m.rs

```diff
@@ -570,7 +526,7 @@ impl<'probe> CoreInterface for Armv6m<'probe> {
 
     fn step(&mut self) -> Result<CoreInformation, Error> {
         // First check if we stopped on a breakpoint, because this requires special handling before we can continue.
-        let pc_before_step = self.read_core_reg(self.registers().program_counter().id)?;
+        let pc_before_step = self.read_core_reg(self.registers().program_counter()?.id)?;
```

Comment:
Why is this an `id` _field_ when we also have an `id` _method_?

Reply:
@Yatekii Good catch. It should use the id() method, but as you pointed out elsewhere, this should be done implicitely with an `Into`. I will fix it :)

Reply:
@Yatekii Good catch. It should use the id() method, but as you pointed out elsewhere, this should be done implicitely with an `Into`. I will fix it :)

Reply:
@Yatekii On further investigation, there is already an `Into` implementation in `core::registers`, but apparently Rust won't invoke it implicitly after the `?`. You can do it explicitly by adding a `.into()`. I'm going to settle for just adding `.id()` because I find it easier to read.